### ISSUE TITLE
Budget with multiple TagKeyValue cost filters

### ISSUE THREAD

**Issue by dmrzzz** [hashicorp/terraform-provider-aws#5890]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I want to create a Budget with more than one TagKeyValue in cost_filters.  AWS supports this, but AFAICT Terraform currently does not.

### New or Affected Resource(s)

* aws_budgets_budget

### Potential Terraform Configuration

```hcl
provider "aws" {
  region = "us-east-2"
}

resource "aws_budgets_budget" "test" {
  name              = "test"
  budget_type       = "COST"
  limit_amount      = "500"
  limit_unit        = "USD"
  time_unit         = "MONTHLY"
  time_period_start = "2018-01-01_00:00"

  cost_filters {
    TagKeyValue = ["user:Service$IP Address Management", "user:Environment$Dev"]
  }

  cost_types {
    use_amortized = true
  }
}
```

Currently (Terraform v0.11.8, provider.aws v1.36.0) this fails with
```
Error: aws_budgets_budget.test: cost_filters (TagKeyValue): '' expected type 'string', got unconvertible type '[]interface {}'
```

### etc

Just for fun, let's comment out the TagKeyValue line above and then `terraform apply` (creating a Budget with no cost_filters).  Now I can add my desired filters manually using the AWS web console.

At this point, we can see that AWS CLI returns an array for TagKeyValue (hence my belief that Terraform ought to accept an array for it):

```
$ aws budgets describe-budget --account-id 378517677616 --budget-name test
{
    "Budget": {
        "BudgetType": "COST",
        "TimePeriod": {
            "End": 3706473600.0,
            "Start": 1535760000.0
        },
        "CalculatedSpend": {
            "ActualSpend": {
                "Unit": "USD",
                "Amount": "0"
            }
        },
        "CostTypes": {
            "UseBlended": false,
            "IncludeCredit": true,
            "IncludeSubscription": true,
            "IncludeRecurring": true,
            "IncludeUpfront": true,
            "UseAmortized": true,
            "IncludeSupport": true,
            "IncludeDiscount": true,
            "IncludeRefund": true,
            "IncludeTax": true,
            "IncludeOtherSubscription": true
        },
        "BudgetName": "test",
        "BudgetLimit": {
            "Unit": "USD",
            "Amount": "500"
        },
        "TimeUnit": "MONTHLY",
        "CostFilters": {
            "TagKeyValue": [
                "user:Service$IP Address Management",
                "user:Environment$Dev"
            ]
        }
    }
}
```

**Comment by mozamimy** [comment]
I'm interested in working on this issue. 

I think other keys of `cost_filters` include the same problem. For example, `Region`, `LinkedAccount`, etc.

I'm going to create a pull request to fix it. However, it may take a while since I'm a beginner of terraform-provider-aws's implementation.

**Comment by ButterflyServiceDesk** [comment]
Just to add to this I have been trying to create a simple budget as well that filters for multiple linked accounts. Below is an example:
`cost_filters = {
    #LinkedAccount = "111111111,222222222,3333333333,444444444,555555555"
  }`

Now when you apply this it actually creates a filter for an account "111111111,222222222,3333333333,444444444,555555555" and not a filter for each account.  I have  also tried it with spaces after the commas. 

Even did a test were I manually created a budget in AWS, added the filter and then did a terraform import. Copied the filter from the state file and used that to create a new budget and still no dice.

Do we have any idea when this will be fixed considering it was originally reported in april?

Regards

**Comment by jfortier** [comment]
The proper way to do this is just to add multiple instances of the TagKeyValues property like so:

```
cost_filters {
    TagKeyValue = "user:Service$IP Address Management"
    TagKeyValue = "user:Environment$Dev"
}
```

I just decided to try this on whim, and it works for me – although using key-values in line with my setup. Either way, changing the environment parameter from staging to production showed two different budgets, and only for the project tag I selected. This allows me to set budgets on a per project and per environment basis. Excellent!
